app/src/main/java/org/mozilla/tv/firefox/ScreenControllerStateMachine.kt

```diff
@@ -0,0 +1,88 @@
+
+// TODO add this stuff
+
+package org.mozilla.tv.firefox
+
+import android.support.annotation.VisibleForTesting
+
+/**
+ * TODO
+ */
+class ScreenControllerStateMachine(private val currentUrlIsHome: () -> Boolean) {
+
+    enum class ActiveScreen {
+        NAVIGATION_OVERLAY, WEB_RENDER, POCKET, SETTINGS
+    }
+
+    enum class Transition {
+        ADD_OVERLAY, REMOVE_OVERLAY, ADD_POCKET, REMOVE_POCKET, ADD_SETTINGS, REMOVE_SETTINGS, EXIT_APP, NO_OP
+    }
+
+    var currentActiveScreen: ActiveScreen = ActiveScreen.NAVIGATION_OVERLAY
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        set
+
+    fun menuPress(): Transition {
+        return when (currentActiveScreen) {
+            ActiveScreen.NAVIGATION_OVERLAY -> {
+                return if (currentUrlIsHome()) {
+                    Transition.NO_OP
+                } else {
+                    currentActiveScreen = ActiveScreen.WEB_RENDER
+                    Transition.REMOVE_OVERLAY
+                }
+            }
+            ActiveScreen.WEB_RENDER -> {
+                currentActiveScreen = ActiveScreen.NAVIGATION_OVERLAY
+                Transition.ADD_OVERLAY
+            }
+            ActiveScreen.POCKET -> Transition.NO_OP
+            ActiveScreen.SETTINGS -> Transition.NO_OP
+        }
+    }
+
+    fun backPress(): Transition {
+        return when (currentActiveScreen) {
+            ActiveScreen.NAVIGATION_OVERLAY -> {
+                return if (currentUrlIsHome()) {
```

Comment:
Yeah, this is in our known bugs (see emails), but since we don't open the app through an Intent right now (Youtube casting would do this but we don't have that), Ashley was fine postponing that fix.

Reply:
@liuche This happens when you open the app from the youtube.com app and could potentially happen from other apps on the Fire TV device. Since none of us use Fire TV regularly, without telemetry, it's hard to say how often this use case occurs.